src/ADP.Portal.Api/Services/AdoRestAPIService.cs

```diff
@@ -0,0 +1,61 @@
+﻿using ADP.Portal.Api.Config;
+using ADP.Portal.Api.Models.Ado;
+using ADP.Portal.Core.Ado.Infrastructure;
+using Microsoft.Extensions.Options;
+using System.Net.Http.Headers;
+
+namespace ADP.Portal.Api.Services
+{
+    public class AdoRestApiService : IAdoRestApiService
+    {
+        private readonly ILogger<AdoRestApiService> logger;
+        private readonly string adoOrgUrl;
+        private readonly HttpClient client;
+
+        public AdoRestApiService(ILogger<AdoRestApiService> logger, IOptions<AdoConfig> configuration, HttpClient client)
+        {
+            this.logger = logger;
+            adoOrgUrl = configuration.Value.OrganizationUrl;
+            this.client = client;
+            client.DefaultRequestHeaders.Accept.Add(
+                    new MediaTypeWithQualityHeaderValue("application/json"));
+
+            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic",
+                Convert.ToBase64String(
+                    System.Text.Encoding.ASCII.GetBytes(
+                        string.Format("{0}:{1}", "", configuration.Value.PatToken))));
+        }
+
+
+
+        public async Task<string> GetUserIdAsync(string projectName, string userName)
+        {
+            var uri = adoOrgUrl.Replace("dev.azure.com", "vssps.dev.azure.com") + "/_apis/identities?searchFilter=General&filterValue=[" + projectName + "]\\" + userName + "&queryMembership=None&api-version=7.1-preview.1";
```

Comment:
there no libraries for these calls?

Reply:
As discussed there is no sdk available for securityrole.